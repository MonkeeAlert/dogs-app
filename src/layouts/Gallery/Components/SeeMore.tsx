import {useNavigation} from '@react-navigation/native';
import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {useDispatch} from 'react-redux';
import {useTheme} from '../../../assets/theme';
import {DefaultButton} from '../../../components/buttons';
import {toggleInHistory} from '../../../redux/actions/listActions';
import {colors, dogs, text} from '../../../utils/constants';
import {parseDog} from '../../../utils/functions';
import {ExtendedNavigationProp} from '../../../utils/types';

interface Props {
  search: string;
}

export const SeeMore = ({search}: Props) => {
  const {mode} = useTheme();
  const dispatch = useDispatch();
  const {navigate} =
    useNavigation<
      ExtendedNavigationProp<'CatalogTabs', {params: {search: string}}>
    >();

  const handlePress = () => {
    const name = parseDog(search);
    navigate('CatalogTabs', {
      screen: 'Catalog',
      params: {
        search: name,
      },
    });
    dispatch(toggleInHistory({name, img: dogs[search].img as string}));
  };

  return (
    <View style={styles.container}>
      <DefaultButton color={colors.turquoise} onPress={handlePress}>
        <Text style={{...styles.text, color: mode.card}}>See more</Text>
      </DefaultButton>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 49,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    fontSize: text.s,
    fontWeight: '500',
  },
});
