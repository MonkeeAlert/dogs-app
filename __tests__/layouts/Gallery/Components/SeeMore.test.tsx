import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {create} from 'react-test-renderer';
import SeeMore from '../../../../src/layouts/Gallery/Components/SeeMore';
import {mockedNavigate} from '../../../../jest.setup';

describe('GalleryModal (SeeMore)', () => {
  it('matches snapshot', () => {
    const tree = create(<SeeMore search="foo" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('navigates to another screen', () => {
    const {getByTestId} = render(<SeeMore search={'foo'} />);
    const btn = getByTestId('DefaultButton_TouchableOpacity');

    act(() => {
      fireEvent.press(btn);
      expect(mockedNavigate).toHaveBeenCalled();
    });
  });
});
