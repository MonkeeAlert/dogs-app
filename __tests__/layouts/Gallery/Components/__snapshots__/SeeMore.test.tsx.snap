// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryModal (SeeMore) matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 49,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="DefaultButton_TouchableOpacity"
  >
    <View
      style={
        Object {
          "backgroundColor": "#2BD9A5",
          "borderRadius": 7,
          "margin": 3.5,
          "padding": 10,
        }
      }
      testID="DefaultButton_View"
    >
      <Text
        style={
          Object {
            "color": "#fffafb",
            "fontSize": 14,
            "fontWeight": "500",
          }
        }
      >
        See more
      </Text>
    </View>
  </View>
</View>
`;
